src/plugins/uploader.js

```diff
@@ -42,39 +42,37 @@ class Uploader {
 		});
 
 		uploader.on("complete", (data) => {
-			handleSaving(data).then((randomName) => {
-				const randomFileName = randomName;
-				const slug = data.file.base;
-				const url = `uploads/${randomFileName}/${slug}`;
+			let randomName;
+			let destPath;
+
+			// Generate a random file name for storage on disk
+			do {
+				randomName = crypto.randomBytes(8).toString("hex");
+				destPath = path.join(Helper.getFileUploadPath(), randomName.substring(0, 2), randomName);
+			} while (fs.stat(destPath, (err) => (err ? true : false)));
+
+			fsextra.move(data.file.pathName, destPath).then(() => {
+				const slug = path.basename(data.file.pathName);
+				const url = `uploads/${randomName}/${slug}`;
 				client.emit("upload:success", url);
+			}).catch(() => {
+				log.warn(`Unable to move uploaded file "${data.file.pathName}"`);
+
+				// Emit failed upload to the client if file move fails
+				socket.emit("siofu_error", {
```

Comment:
What do we do with this error? I'm not seeing any handling of `siofu_error` on the client, is it specific to the file upload library or something?
If it's for own consumption, maybe `file_upload_error` instead?

Reply:
Client library emits error event which our code hsndles.

Reply:
Yeah, but this is our code emitting the error to our client, no?

Reply:
Yeah, but I didn't want to write our own error handler here because this error should basically never happen.